Content.Server/GameObjects/Components/Damage/DamageOnToolInteractComponent.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Content.Server.GameObjects.Components.Interactable;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.Components.Interactable;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.Damage
+{
+    [RegisterComponent]
+    class DamageOnToolInteractComponent : Component, IInteractUsing
+    {
+        public override string Name => "DamageOnToolInteract";
+
+        /* Set in YAML */
+        protected int Damage;
+        private List<ToolQuality> _tools = new List<ToolQuality>();
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref Damage, "damage", 0);
+
+            serializer.DataField(ref _tools, "tools", new List<ToolQuality>());
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            Owner.EnsureComponent<DamageableComponent>();
+        }
+
+        public bool InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            
+
+            if (eventArgs.Using.TryGetComponent<ToolComponent>(out var tool))
+            {
+                if (tool.HasQuality(ToolQuality.Welding))
+                {
+                    if (eventArgs.Using.TryGetComponent<WelderComponent>(out WelderComponent welder))
```

Comment:
This makes it so that the `tools` list is ignored if the tool in question is a welder, which doesn't seem very correct.

Reply:
First of all I'm sorry for the late reply, I've been rather busy lately and haven't had the time I would have liked to contribute. I've corrected this mistake.

Reply:
@PJB3005 Its now 100% fixed. I've tested it. the expression toolQuality == ToolQuality.Welding should fix the issue you were referring to. That way the specific welder function should only fire if the tool is both a welder and the object can take damage from a welder.